### PR TITLE
[Docs] Recommend trait usage in Plugin Development Guide

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | 1.6
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

For a long time already, we provide all entities [overridden in Sylius-Standard](https://github.com/Sylius/Sylius-Standard/tree/master/src/Entity). This approach enforces annotations mapping by default but allows Sylius to use multiple plugins extending the same entity at the same time. 

We should definitely recommend this approach in our **Plugin Development Guide**.

### CODE REVIEW

**Comment by JoppeDC** [comment]
Thanks for the further information @Zales0123 , this definetly clears up some misunderstanding.

**Comment by lchrusciel** [comment]
PS. Nice PR number 🎉

File: docs/plugin-development-guide/implementation.rst [hunk 1]
```diff
@@ -86,54 +54,23 @@ PHPSpec, PHPUnit, or any other unit testing tool):
...
-            availableOnDemand:
-                type: boolean
-
-Then our new entity should be configured as a resource model:
-
-.. code-block:: yaml
-
-    # src/Resources/config/config.yml
-
-    sylius_product:
```
Comment (lchrusciel) [inline docs/plugin-development-guide/implementation.rst]: What do you think about recommending this configuration in the test folder, so the entity will be checked if wired up properly (and preferably will be a good introduction for testing)

**Comment by lchrusciel** [comment]
Thank you, Mateusz! :tada:

**Pull request merged**